app.go

```diff
@@ -130,9 +153,39 @@ func (s *App) start(funcs ...interface{}) error {
 	return nil
 }
 
+// Invoke calls the given functions in-order with objects from the D.I.
+// container. The function blocks until all functions have executed.
+//
+// Functions may optionally return errors. The error returned by the first
+// failing function will be returned to the caller as-is.
+//
+// This function MUST NOT be called before Start or after Stop.
+func (s *App) Invoke(funcs ...interface{}) error {
```

Comment:
We definitely need to be able to return an error. The idea is that the invoked function can return an error:

```
app.Invoke(func(d *yarpc.Dispatcher) error {
  if err := doSomethingWith(d); err != nil { return err }
  doSomethingElse()
})
```